src/Util.h

```diff
@@ -290,6 +290,17 @@ inline bool isNumber(const char* str) {
 //                           General Container Utilities
 // -------------------------------------------------------------------------------
 
+template <typename A>
+using Own = std::unique_ptr<A>;
+
+template <typename A>
+using VecOwn = std::vector<Own<A>>;
+
+template <typename A, typename B = A, typename... Args>
+Own<A> mk(Args&&... xs) {
```

Comment:
`mk` is not a helpful name. A more descriptive name would be good. As it is, I'm not sure what the function is for. A brief description would also help.

From the use in other files it seems to just be a way to shorten `std::unique_ptr` and `std::vector`, which makes it less clear what they are. Is there another advantage to these templates?

Reply:
'mk' is shorthand for 'make'  (common idiom from Haskell leaking through), which given the ubiquity of the operation seemed to be appropriate to make quite short/easy to type/read. It's almost identical to `std::make_unique` except that you can optionally specify what supertype to use for the pointer parameterisation (useful when this interacts with type deduction, e.g. `auto`, lambdas, and template helpers).

Reply:
As for 'just to shorten it': It is. However, given how common this particular pattern is I find it more tiresome to read in its entirety (common things should have short names, much like words in natural languages, `std::vector<std::unique_ptr<A>>` vs `VecOwn<A>`). If this is a point of contention I'll revert.